File: Gabarito/JavaScript/1 - Cadastro de Alunos.js
/*# Cadastro de Alunos
Crie um programa em que seja possível cadastrar alunos, armazenando nome, idade e nota. Em seguida, exiba a média das notas dos alunos cadastrados.*/

class Aluno {
    constructor(nome, idade, nota) {
      this.nome = nome;
      this.idade = idade;
      this.nota = nota;
    }
  }
  
  function calcularMediaAlunos(alunos) {
    let somaNotas = 0;
    for (let aluno of alunos) {
      somaNotas += aluno.nota;
    }
    const media = somaNotas / alunos.length;
    return media;
  }
  
  const alunos = [];
  alunos.push(new Aluno("João", 18, 8.5));
  alunos.push(new Aluno("Maria", 19, 9.0));
  alunos.push(new Aluno("Pedro", 20, 7.5));
  
  const mediaDasNotas = calcularMediaAlunos(alunos);
  console.log(`A média das notas dos alunos é: ${mediaDasNotas}`);
  

File: Gabarito/Python/1 - Cadastro de Alunos.py
# Cadastro de Alunos
# Crie um programa em que seja possível cadastrar alunos, armazenando nome, idade e nota. Em seguida, exiba a média das notas dos alunos cadastrados.*/

class Aluno:
    def __init__(self, nome, idade, nota):
        self.nome = nome
        self.idade = idade
        self.nota = nota

def calcular_media_alunos(alunos):
    soma_notas = sum(aluno.nota for aluno in alunos)
    media = soma_notas / len(alunos)
    return media

alunos = []
alunos.append(Aluno("João", 18, 8.5))
alunos.append(Aluno("Maria", 19, 9.0))
alunos.append(Aluno("Pedro", 20, 7.5))

media_das_notas = calcular_media_alunos(alunos)
print(f'A média das notas dos alunos é: {media_das_notas}')
